jetty-quickstart/src/main/java/org/eclipse/jetty/quickstart/QuickStartConfiguration.java

```diff
@@ -160,37 +178,34 @@ protected void configure(QuickStartGeneratorConfiguration generator, WebAppConte
     @Override
     public void configure(WebAppContext context) throws Exception
     {
-        if (!_quickStart)
-        {
-            super.configure(context);
-        }
-        else
-        {
-            //add the processor to handle normal web.xml content
-            context.getMetaData().addDescriptorProcessor(new StandardDescriptorProcessor());
+        Resource quickStartWebXml = getQuickStartWebXml(context);
+        // Don't run configure() if quickstart-web.xml does not exist
+        if (quickStartWebXml == null || !quickStartWebXml.exists())
+            return;
 
-            //add a processor to handle extended web.xml format
-            context.getMetaData().addDescriptorProcessor(new QuickStartDescriptorProcessor());
+        // add the processor to handle normal web.xml content
+        context.getMetaData().addDescriptorProcessor(new StandardDescriptorProcessor());
 
-            //add a decorator that will find introspectable annotations
-            context.getObjectFactory().addDecorator(new AnnotationDecorator(context)); //this must be the last Decorator because they are run in reverse order!
+        // add a processor to handle extended web.xml format
+        context.getMetaData().addDescriptorProcessor(new QuickStartDescriptorProcessor());
 
-            //add a context bean that will run ServletContainerInitializers as the context starts
-            ServletContainerInitializersStarter starter = (ServletContainerInitializersStarter)context.getAttribute(AnnotationConfiguration.CONTAINER_INITIALIZER_STARTER);
-            if (starter != null)
-                throw new IllegalStateException("ServletContainerInitializersStarter already exists");
-            starter = new ServletContainerInitializersStarter(context);
-            context.setAttribute(AnnotationConfiguration.CONTAINER_INITIALIZER_STARTER, starter);
-            context.addBean(starter, true);
+        // add a decorator that will find introspectable annotations
+        context.getObjectFactory().addDecorator(new AnnotationDecorator(context)); //this must be the last Decorator because they are run in reverse order!
 
-            LOG.debug("configured {}", this);
-        }
+        // add a context bean that will run ServletContainerInitializers as the context starts
+        ServletContainerInitializersStarter starter = (ServletContainerInitializersStarter)context.getAttribute(AnnotationConfiguration.CONTAINER_INITIALIZER_STARTER);
+        if (starter != null)
+            throw new IllegalStateException("ServletContainerInitializersStarter already exists");
+        starter = new ServletContainerInitializersStarter(context);
+        context.setAttribute(AnnotationConfiguration.CONTAINER_INITIALIZER_STARTER, starter);
+        context.addBean(starter, true);
+
+        LOG.debug("configured {}", this);
     }
 
     @Override
     public void postConfigure(WebAppContext context) throws Exception
     {
-        super.postConfigure(context);
```

Comment:
All `Configuration` classes call their `super()` methods, so leave it in.

Reply:
That's an incorrect assumption.
And the super calls are no-ops.

super.preConfigure(context)

- AnnotationConfiguration (overridden only)
- EnvConfiguration (overridden only)
- PlusConfiguration (overridden only)
- FragmentConfiguration (overridden only)
- MetaInfConfiguration (overridden only)
- WebInfConfiguration (overridden only)
- WebXmlConfiguration (overridden only)
- QuickStartConfiguration (overridden + calls super)
- OSGIMetaInfConfiguration (overridden + calls super)
- QuickStartGeneratorConfiguration (overridden + calls super)

super.configure(context)

- AnnotationConfiguration (overridden only)
- EnvConfiguration (overridden only)
- PlusConfiguration (overridden only)
- QuickStartGeneratorConfiguration (overridden only)
- JettyWebXmlConfiguration (overridden only)
- MetaInfConfiguration (overridden only)
- WebXmlConfiguration (overridden only)
- QuickStartConfiguration (overridden + calls super)
- AntWebInfConfiguration (overridden + calls super)
- MavenWebInfConfiguration (overridden + calls super)
- OSGIMetaInfConfiguration (overridden + calls super)

super.postConfigure(context)

- AnnotationConfiguration (overridden only)
- PlusConfiguration (overridden only)
- FragmentConfiguration (overridden only)
- MetaInfConfiguration (overridden only)
- OSGIMetaInfConfiguration (overridden + calls super)
- AnnotationConfiguration (overridden + calls super)
- QuickStartConfiguration (overridden + calls super)